kartothek/core/dataset.py

```diff
@@ -311,6 +311,20 @@ def query(self, indices: List[IndexBase] = None, **kwargs) -> List[str]:
         for column, value in kwargs.items():
             if column in combined_indices:
                 candidate_set &= set(combined_indices[column].query(value))
+            else:
+                if (
+                    # FIXME: `self.tables[0]`. What if column exists in another table?
+                    not self.table_meta[self.tables[0]]
```

Comment:
As long as we support multi table datasets we can not make this assumption. You'd need to check all tables